lib/checks/aria/allowed-role.js

```diff
@@ -0,0 +1,207 @@
+		'option', 'radio', 'switch', 'tab'
+	],
+	'details': ['alertdialog'],
+	'dl': ['group', 'presentation', 'none', 'doc-glossary'],
+	'embed': [ 'application', 'document', 'presentation', 'none', 'img' ],
+	'figcaption': [ 'group', 'presentation', 'none' ],
+	'fieldset': 	[ 'group', 'presentation', 'none' ],
+	'footer': [ 'group', 'none', 'presentation', 'doc-footnote' ],
+	'form': [ 'search', 'none', 'presentation' ],
+	'h1': [ 'tab', 'none', 'presentation', 'doc-subtitle' ],
+	'h2': [ 'tab', 'none', 'presentation', 'doc-subtitle' ],
+	'h3': [ 'tab', 'none', 'presentation', 'doc-subtitle' ],
+	'h4': [ 'tab', 'none', 'presentation', 'doc-subtitle' ],
+	'h5': [ 'tab', 'none', 'presentation', 'doc-subtitle' ],
+	'h6': [ 'tab', 'none', 'presentation', 'doc-subtitle' ],
+	'header': [ 'group', 'none', 'presentation', 'doc-footnote' ],
+	'hr': [ 'presentation', 'doc-pagebreak' ],
+	'iframe': [ 'application', 'document', 'img' ],
+	'imgWithEmptyAlt': [ 'presentation', 'none' ],
+	'inputTypeButton': [
+		'link, menuitem', 'menuitemcheckbox', 'menuitemradio', 'radio', 'switch',
+		'option', 'tab'
+	],
+	'inputTypeImage': [
+		'link', 'menuitem', 'menuitemcheckbox', 'menuitemradio', 'radio', 'switch'
+	],
+	'inputTypeCheckbox': [ 'button', 'menuitemcheckbox', 'option', 'switch' ],
+	'li': [
+		'menuitem', 'menuitemcheckbox', 'menuitemradio', 'option', 'none',
+		'presentation', 'radio', 'separator', 'tab', 'treeitem', 'doc-biblioentry',
+		'doc-endnote'
+	],
+	'nav': [ 'doc-index', 'doc-pagelist', 'doc-toc' ],
+	'object': [ 'application', 'document', 'img' ],
+	'ol': [
+		'directory', 'group', 'listbox', 'menu', 'menubar,none', 'presentation ',
+		'radiogroup', 'tablist', 'toolbar', 'tree'
+	],
+	'section': [
+		'alert', 'alertdialog', 'application', 'banner', 'complementary',
+		'contentinfo', 'dialog', 'document', 'feed', 'log', 'main', 'marquee',
+		'navigation', 'none', 'presentation', 'search', 'status', 'tabpanel',
+		'doc-abstract', 'doc-acknowledgments', 'doc-afterword', 'doc-appendix',
+		'doc-bibliography', 'doc-chapter', 'doc-colophon', 'doc-conclusion',
+		'doc-credit', 'doc-credits', 'doc-dedication', 'doc-endnotes', 'doc-epilogue',
+		'doc-errata', 'doc-example', 'doc-foreword', 'doc-index', 'doc-introduction',
+		'doc-notice', 'doc-pagelist', 'doc-part', 'doc-preface', 'doc-prologue',
+		'doc-pullquote', 'doc-qna', 'doc-toc'
+	],
+	'svg': [ 'application', 'document', 'img' ],
+	'ul': [
+		'directory', 'group', 'listbox', 'menu', 'menubar', 'radiogroup',
+		'tablist', 'toolbar', 'tree', 'presentation'
+	],
+	'video': [ 'application' ]
+};
+
+/**
+ * Finds nearest parent with a specifig tagName
+ * @param  {HTMLElement} 		el      		child - starting pointer
+ * @param  {Array<String>} 	tagName 		Array containg capatilized tagnames
+ * @return {HTMLElement}         				Parent that matches one of the tagnames
+ */
+function getParentWithTagName(el, tagName) {
+	while (el.parentNode){
```

Comment:
There's some inconsistent spacing around the while and if declarations here, as well as the comment blocks. Maybe we should add an automated tool to clean that stuff up. WDYT about adding [Prettier](https://github.com/prettier/prettier) in a separate PR, Wilco?

Reply:
I'm open to experimenting with Prettier. It seemed pretty cool. I don't think we should start with axe-core for that. Maybe a less impactful project to try it out on and adopt it into others if we like it.